mmv1/products/compute/FirewallPolicyRule.yaml

```diff
@@ -0,0 +1,249 @@
+# Copyright 2024 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'FirewallPolicyRule'
+kind: 'compute#firewallPolicyRule'
+base_url: 'locations/global/firewallPolicies/{{firewall_policy}}'
```

Comment:
The error comes from having the url being nested in itself.

From the logs we can see:
```hcl
---[ REQUEST ]---------------------------------------
POST /compute/beta/locations/global/firewallPolicies/locations/global/firewallPolicies/79416795910/addRule?alt=json HTTP/1.1
Host: compute.googleapis.com
User-Agent: Terraform/1.8.3 (+https://www.terraform.io) Terraform-Plugin-SDK/2.33.0 terraform-provider-google-beta/acc
Content-Length: 458
Content-Type: application/json
Accept-Encoding: gzip

{
 "action": "deny",
 "description": "Resource created for Terraform acceptance testing",
 "direction": "INGRESS",
 "match": {
  "layer4Configs": [
   {
    "ipProtocol": "tcp",
    "ports": [
     "80",
     "8080"
    ]
   },
   {
    "ipProtocol": "all"
   }
  ],
  "srcAddressGroups": [
   "organizations/529579013760/locations/global/addressGroups/tf-test-policykir3yqimoi"
  ],
  "srcFqdns": [
   "google.com"
  ],
  "srcIpRanges": [
   "11.100.0.1/32"
  ],
  "srcRegionCodes": [
   "US"
  ],
  "srcThreatIntelligences": [
   "iplist-known-malicious-ips"
  ]
 },
 "priority": 9001
}

-----------------------------------------------------
2024/08/09 23:44:32 [DEBUG] Google API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 404 Not Found
Content-Length: 1670
Content-Type: text/html; charset=UTF-8
Date: Fri, 09 Aug 2024 23:44:32 GMT
Server: ESF
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0
```

What to look for is the POST url:`POST /compute/beta/locations/global/firewallPolicies/locations/global/firewallPolicies/79416795910/addRule`

Where `{{firewall_policy}} = locations/global/firewallPolicies/79416795910` in `locations/global/firewallPolicies/{{firewall_policy}}`

tl;dr you'll want to just use `{{firewall_policy}}` for the urls here instead of what's set currently. Let me know if something doesn't make sense from this explanation.

Reply:
Oh, I did the fix for this error but I forgot to push the changes.
I'll push in a couple of minutues!
Thanks